core/common/test/LocalDateRangeTest.kt

```diff
@@ -0,0 +1,312 @@
+        assertContentEquals(
+            (1..5).map { LocalDate(2000, 1, it) },
+            Jan_01_2000..Jan_05_2000
+        )
+        assertContentEquals(
+            (1..<5).map { LocalDate(2000, 1, it) },
+            Jan_01_2000..<Jan_05_2000
+        )
+        assertTrue {
+            (LocalDate.MIN..<LocalDate.MIN).isEmpty()
+        }
+        assertTrue {
+            (LocalDate.MIN..<LocalDate.MAX).isNotEmpty()
+        }
+        assertContentEquals(
+            listOf(Jan_01_2000),
+            Jan_01_2000..Jan_01_2000
+        )
+        assertContentEquals(
+            listOf(
+                LocalDate(1999, 12, 30),
+                LocalDate(1999, 12, 31),
+                LocalDate(2000, 1, 1),
+                LocalDate(2000, 1, 2)
+            ),
+            Dec_30_1999..Jan_02_2000
+        )
+    }
+
+    @Test
+    fun backwardRange() {
+        assertContentEquals(
+            (5 downTo 1).map { LocalDate(2000, 1, it) },
+            Jan_05_2000 downTo Jan_01_2000
+        )
+        assertContentEquals(
+            listOf(Jan_01_2000),
+            Jan_01_2000 downTo Jan_01_2000
+        )
+        assertContentEquals(
+            listOf(
+                LocalDate(2000, 1, 2),
+                LocalDate(2000, 1, 1),
+                LocalDate(1999, 12, 31),
+                LocalDate(1999, 12, 30),
+            ),
+            Jan_02_2000 downTo Dec_30_1999
+        )
+    }
+
+    @Test
+    fun step() {
+        assertContentEquals(
+            (1..24 step 2).map { LocalDate(2000, 1, it) },
+            (Jan_01_2000..Jan_24_2000).step(2, DateTimeUnit.DAY)
+        )
+        assertContentEquals(
+            (24 downTo 1  step 2).map { LocalDate(2000, 1, it) },
+            (Jan_24_2000 downTo Jan_01_2000).step(2, DateTimeUnit.DAY)
+        )
+        assertContentEquals(
+            (Jan_01_2000..Jan_24_2000).step(Long.MAX_VALUE / 2, DateTimeUnit.WEEK),
+            listOf(Jan_01_2000)
+        )
+    }
+
+    @Test
+    fun string() {
+        assertEquals(
+            "2000-01-01..2000-01-05",
+            (Jan_01_2000..Jan_05_2000).toString()
+        )
+        assertEquals(
+            "2000-01-05 downTo 2000-01-01 step -1D",
+            (Jan_05_2000 downTo Jan_01_2000).toString()
+        )
+        assertEquals(
+            "2000-01-01..2000-01-05 step 1D",
+            LocalDateProgression.fromClosedRange(Jan_01_2000, Jan_05_2000, 1, DateTimeUnit.DAY).toString()
+        )
+        assertEquals(
+            "2000-01-05 downTo 2000-01-01 step -1D",
+            LocalDateProgression.fromClosedRange(Jan_05_2000, Jan_01_2000, -1, DateTimeUnit.DAY).toString()
+        )
+    }
+
+    @Test
+    fun random() {
+        assertEquals(
+            Jan_01_2000,
+            (Jan_01_2000..Jan_01_2000).random()
+        )
+
+        assertEquals(
+            Jan_01_2000,
+            (Jan_01_2000 downTo Jan_01_2000).random()
+        )
+
+        assertFailsWith<IllegalArgumentException> {
```

Comment:
Should be `NoSuchElementException`

Reply:
The underlying `Random.nextLong(range: LongRange)` call throws an `IllegalArgumentException` for an empty range. Do we want to wrap that call in a try-catch and rethrow as a `NoSuchElementException`?

Reply:
Yes, that's what `random()` function throws on empty collections and lists in stdlib.

Reply:
But it would be better to check for an empty range instead of wrapping a call to `nextLong()`